pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -59,10 +61,15 @@ type AdvertisableAddressResolver interface {
 	Resolve(observedAdddress ma.Multiaddr) (ma.Multiaddr, error)
 }
 
+type swapBackend interface {
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+}
+
 // Service can perform initiate or handle a handshake between peers.
 type Service struct {
 	signer                crypto.Signer
 	advertisableAddresser AdvertisableAddressResolver
+	swapBackend           swapBackend
```

Comment:
Perhaps it is time to reconsider this name, given that we use it now for settlements, postage stamps and (upcoming) spoofless overlay protection. @ralph-pichler , wdyt?
It's not related to this PR though